apps/outline/docker-compose.yml

```diff
@@ -0,0 +1,95 @@
+version: "3.8"
+
+services:
+  outline:
+    container_name: outline-outline
+    image: docker.getoutline.com/outlinewiki/outline:0.75.2
+    restart: unless-stopped
+    environment:
+      - "DATABASE_URL=postgres://outline:${PG_PASSWORD}@postgres:5432/outline"
+      - "REDIS_URL=redis://redis:6379"
+      - "SECRET_KEY=${OUTLINE_SECRET_KEY}"
+      - "UTILS_SECRET=${OUTLINE_UTILS_SECRET}"
+      - "URL=https://${APP_DOMAIN}"
+      - "FORCE_HTTPS=false"
+      - "FILE_STORAGE=local"
+      - "FILE_STORAGE_UPLOAD_MAX_SIZE=26214400"
+      - "PGSSLMODE=disable"
+      - "OIDC_CLIENT_ID=${OIDC_CLID}"
+      - "OIDC_CLIENT_SECRET=${OIDC_CLKEY}"
+      - "OIDC_AUTH_URI=https://${APP_DOMAIN}/uc/oauth/authorize/"
+      - "OIDC_TOKEN_URI=https://${APP_DOMAIN}/uc/oauth/token/"
+      - "OIDC_USERINFO_URI=https://${APP_DOMAIN}/uc/oauth/userinfo/"
+      - "OIDC_USERNAME_CLAIM=preferred_username"
+      - "OIDC_DISPLAY_NAME=OpenID"
+      - "OIDC_SCOPES=openid profile email"
+    depends_on:
+      - postgres
+      - redis
+    networks:
+      - tipi_main_network
+
+  postgres:
+    container_name: outline-postgres
+    image: postgres:16-alpine
+    restart: unless-stopped
+    environment:
+      - "POSTGRES_USER=outline"
+      - "POSTGRES_PASSWORD=${PG_PASSWORD}"
+      - "POSTGRES_DB=outline"
+    volumes:
+      - ${APP_DATA_DIR}/data/pgdata:/var/lib/postgresql/data
+    networks:
+      - tipi_main_network
+
+  redis:
+    container_name: outline-redis
+    image: redis:7-alpine
+    restart: unless-stopped
+    networks:
+      - tipi_main_network
+
+  oidc:
+    container_name: outline-oidc
+    image: ghcr.io/hex-developer/oidc-provider:v0.2.0
```

Comment:
I see that you use your own image here. Is the oidc server needed for the app?

Reply:
I think including the OIDC server is necessary. Otherwise users will have to setup Google or Slack for authentication. A local login method is better for self-hosting. The OIDC server I have used is a fork which I have updated to the latest version. You can report any problems you may have with it to me.

Reply:
Alright it's fine with me. I will test tomorrow to make sure everything is working.